lua/modules/linear_bolzmann_solvers/response_evaluator/response_evaluator.cc

```diff
@@ -0,0 +1,130 @@
+#include "response_evaluator.h"
+#include "modules/linear_boltzmann_solvers/response_evaluator/response_evaluator.h"
+#include "lua/framework/console/console.h"
+
+namespace opensnlua::lbs
+{
+
+RegisterLuaFunctionNamespace(ClearResponseSources, lbs, ClearResponseSources);
+
+int
+ClearResponseSources(lua_State* L)
+{
+  const int num_args = lua_gettop(L);
+  if (num_args != 1)
+    LuaPostArgAmountError(__FUNCTION__, 1, num_args);
+
+  LuaCheckIntegerValue(__FUNCTION__, L, 1);
+
+  // Get the response evaluator
+  const int handle = lua_tonumber(L, 1);
+  auto& response_evaluator =
+    GetStackItem<opensn::lbs::ResponseEvaluator>(object_stack, handle, __FUNCTION__);
+
+  // Clear the sources
+  response_evaluator.ClearForwardSources();
+  return 0;
+}
+
+RegisterWrapperFunctionNamespace(lbs,
+                                 AddResponseBuffers,
+                                 GetResponseBufferSyntax,
+                                 AddResponseBuffers);
+
+InputParameters
+GetResponseBufferSyntax()
+{
+  InputParameters params;
+  params.AddRequiredParameter<size_t>("arg0",
+                                      "Handle to a <TT>lbs::ResponseEvaluator</TT> object.");
+  params.AddRequiredParameterArray("arg1", "Array of blocks for <TT>BufferOptionsBlock</TT>.");
+  params.LinkParameterToBlock("arg1", "lbs::response::BufferOptionsBlock");
+  return params;
+}
+
+ParameterBlock
+AddResponseBuffers(const InputParameters& params)
+{
+  params.RequireParameter("arg0");
+  params.RequireParameter("arg1");
+
+  const auto handle = params.GetParamValue<size_t>("arg0");
+  auto& response_evaluator =
+    GetStackItem<opensn::lbs::ResponseEvaluator>(object_stack, handle, __FUNCTION__);
+
+  const auto buffer_params = params.GetParam("arg1");
+  for (size_t p = 0; p < buffer_params.NumParameters(); ++p)
+  {
+    auto spec = opensn::lbs::ResponseEvaluator::BufferOptionsBlock();
+    spec.AssignParameters(buffer_params.GetParam(p));
+    response_evaluator.SetBufferOptions(spec);
+  }
+
+  return ParameterBlock();
+}
+
+RegisterWrapperFunctionNamespace(lbs,
+                                 AddResponseSources,
+                                 GetResponseSourceSyntax,
+                                 AddResponseSources);
+
+InputParameters
+GetResponseSourceSyntax()
+{
+  InputParameters params;
+  params.SetGeneralDescription("Add sources to the response evaluator.");
+  params.SetDocGroup("LBSResponseEvaluator");
+
+  params.AddRequiredParameter<size_t>("arg0",
+                                      "Handle to a <TT>lbs::ResponseEvaluator</TT> object.");
+  params.AddRequiredParameterBlock("arg1",
+                                   "A block with the syntax of <TT>SourceOptionsBlock</TT>.");
+  params.LinkParameterToBlock("arg1", "lbs::response::SourceOptionsBlock");
```

Comment:
there is 'arg1' here but it is also on the line above in the addrequiredparameterblock. are those strings such dummy names?

Reply:
Yes, they are dummy names. I couldn't for sure tell you what happens under the hood, but this is how it is done for the `lbs::OptionsBlock` in `linear_boltzmann_solver/a_lbs_solver/lbs_solver.cc`. Those are positional arguments, not keyword arguments.